下に引っ張ってtableviewを更新する

alotofwe opened the pull request:
Infinite Scrollはリフレッシュには対応していないとのことなので，自前で実装します．

UIFreshControlで実装します．
# Close条件
- [x] Passed CI
- [x] 1 LGTM
# チェック方法

FeedとProfileにて，下に引っ張ると最新の状態まで更新されることを
シミュレータ or 実機で確認する

alotofwe commented:
現在表示していない，新しいmicropostsの取得方法を考えたのですが，今のAPIだと
1. page=1で叩く
2. 叩いて取得したmicropostsの中で一番小さいidの値(A)より，現在表示しているmicropostsの中で一番大きいidの値(B)の方が小さければpage=(page+1)で叩く

という方法が考えられますが，この方法では
手順1と2の間にフォローしているユーザが新しくmicropostを投稿した場合，idが(A)のポストがpage=2にも現れてしまいます．

そこで，APIにoffset指定ができるようにして，手順2では(A) - 1をoffsetとして，常にpage=1で叩くようにしようと思っています．

gyugyu commented:
> 手順1と2の間にフォローしているユーザが新しくmicropostを投稿した場合，idが(A)のポストがpage=2にも現れてしまいます．

この ID より小さい（もしくは大きい）ものを n 件表示するという API を作ると捗ると思うよ！

alotofwe commented:
> この ID より小さい（もしくは大きい）ものを n 件表示するという API を作ると捗ると思うよ！

ありがとうございます!
指定したIDより小さいほうが一般的な気がしていて，何も考えずにpage=1を叩き続けることができるので，小さいほうにしようと思っています．

Commit:
Add GetLatestFeed and GetLatestMicroposts

Commit:
Add some column and methods

unixTimeCreatedAt column, add method for append array to microposts,
                  lastUpdate method that return last update time by unix
                  time

Commit:
Add addData method for pull to refresh

Commit:
Refresh Feed or Microposts if pull down

Commit:
Use Nyah

ku00 commented:
実機で投稿をして、それをシュミュレータ側が更新してくれるかどうかで動作確認をしました。
問題なさそうです! :ok_woman:

alotofwe commented:
ありがとう! CIが通り次第シュッとマージします
